homeassistant/components/systemmonitor/sensor.py

```diff
@@ -74,185 +82,289 @@ def get_cpu_icon() -> Literal["mdi:cpu-64-bit", "mdi:cpu-32-bit"]:
     return "mdi:cpu-32-bit"
 
 
-@dataclass(frozen=True)
-class SysMonitorSensorEntityDescription(SensorEntityDescription):
-    """Description for System Monitor sensor entities."""
+def get_processor_temperature(
+    entity: SystemMonitorSensor[dict[str, list[shwtemp]]],
+) -> float | None:
+    """Return processor temperature."""
+    return read_cpu_temperature(entity.coordinator.data)
+
+
+def get_process(entity: SystemMonitorSensor[list[psutil.Process]]) -> str:
+    """Return process."""
+    state = STATE_OFF
+    for proc in entity.coordinator.data:
+        try:
+            _LOGGER.debug("process %s for argument %s", proc.name(), entity.argument)
+            if entity.argument == proc.name():
+                state = STATE_ON
+                break
+        except psutil.NoSuchProcess as err:
+            _LOGGER.warning(
+                "Failed to load process with ID: %s, old name: %s",
+                err.pid,
+                err.name,
+            )
+    return state
+
+
+def get_network(entity: SystemMonitorSensor[dict[str, snetio]]) -> float | None:
+    """Return network in and out."""
+    counters = entity.coordinator.data
+    if entity.argument in counters:
+        counter = counters[entity.argument][IO_COUNTER[entity.entity_description.key]]
+        return round(counter / 1024**2, 1)
+    return None
+
+
+def get_packets(entity: SystemMonitorSensor[dict[str, snetio]]) -> float | None:
+    """Return packets in and out."""
+    counters = entity.coordinator.data
+    if entity.argument in counters:
+        return counters[entity.argument][IO_COUNTER[entity.entity_description.key]]
+    return None
+
+
+def get_throughput(entity: SystemMonitorSensor[dict[str, snetio]]) -> float | None:
+    """Return network throughput in and out."""
+    counters = entity.coordinator.data
+    state = None
+    if entity.argument in counters:
+        counter = counters[entity.argument][IO_COUNTER[entity.entity_description.key]]
+        now = dt_util.utcnow()
```

Comment:
Maybe use `time.time` instead since you are having to convert to `total_seconds` anyways

```diff
diff --git a/homeassistant/components/systemmonitor/sensor.py b/homeassistant/components/systemmonitor/sensor.py
index d2b9b3c11ba..b61f8ef6d38 100644
--- a/homeassistant/components/systemmonitor/sensor.py
+++ b/homeassistant/components/systemmonitor/sensor.py
@@ -8,6 +8,7 @@ from functools import lru_cache
 import logging
 import socket
 import sys
+import time
 from typing import Any, Generic, Literal
 
 import psutil
@@ -130,12 +131,10 @@ def get_throughput(entity: SystemMonitorSensor[dict[str, snetio]]) -> float | No
     state = None
     if entity.argument in counters:
         counter = counters[entity.argument][IO_COUNTER[entity.entity_description.key]]
-        now = dt_util.utcnow()
+        now = time.time()
         if entity.value and entity.update_time and entity.value < counter:
             state = round(
-                (counter - entity.value)
-                / 1000**2
-                / (now - (entity.update_time)).total_seconds(),
+                (counter - entity.value) / 1000**2 / (now - (entity.update_time)),
                 3,
             )
         entity.update_time = now
@@ -745,7 +744,7 @@ class SystemMonitorSensor(CoordinatorEntity[MonitorCoordinator[dataT]], SensorEn
         )
         self.argument = argument
         self.value: int | None = None
-        self.update_time: datetime | None = None
+        self.update_time: float | None = None
 
     @property
     def native_value(self) -> StateType | datetime:
```

Better to use `time.monotonic()` as well if you don't need wall clock time (I don't think you do here) as it won't give bad data if time moves backwards because of NTP sync